modules/tensor_mechanics/include/materials/ADAnisotropicReturnCreepStressUpdateBase.h

```diff
@@ -0,0 +1,59 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADGeneralizedRadialReturnStressUpdate.h"
+
+/**
+ * This class provides baseline functionality for anisotropic (Hill-like) plasticity and creep
+ * models based on the stress update material in a generalized radial return framework.
+ */
+class ADAnisotropicReturnCreepStressUpdateBase : public ADGeneralizedRadialReturnStressUpdate
+{
+public:
+  static InputParameters validParams();
+
+  ADAnisotropicReturnCreepStressUpdateBase(const InputParameters & parameters);
+
+  /**
+   * Compute the strain energy rate density for this inelastic model for the current step.
+   * @param stress The stress tensor at the end of the step
+   * @param strain_rate The strain rate at the end of the step
+   * @return The computed strain energy rate density
+   */
+  virtual Real
+  computeStrainEnergyRateDensity(const ADMaterialProperty<RankTwoTensor> & /*stress*/,
+                                 const ADMaterialProperty<RankTwoTensor> & /*strain_rate*/)
+  {
+    mooseError(
+        "The computation of strain energy rate density needs to be implemented by a child class");
+  }
+
```

Comment:
are you moving this up in the hierarchy in your newer PR?

Reply:
Yeah. That newer PR can be updated once this gets in.